src/thread_pool.cpp

```diff
@@ -75,20 +72,20 @@ auto thread_pool::shutdown() noexcept -> void
     }
 }
 
-auto thread_pool::executor(std::stop_token stop_token, std::size_t idx) -> void
+auto thread_pool::executor(std::size_t idx) -> void
 {
     if (m_opts.on_thread_start_functor != nullptr)
     {
         m_opts.on_thread_start_functor(idx);
     }
 
-    while (!stop_token.stop_requested())
+    while (!m_shutdown_requested.load(std::memory_order::relaxed))
```

Comment:
```suggestion
    while (!m_shutdown_requested.load(std::memory_order::acquire))
```

Reply:
In theory the unique lock should make the `acquire` here unnecessary but its possible the `!m_queue.empty()` triggers us out of the condition variable and hen the `m_shutdown_requested` is never refreshed so the `relaxed` could read stale data.  Lets update these two lines and see if it still reproduces please.

Reply:
My best guess at this point is in the test its not exiting the outer loop properly and hangs.

Reply:
I've made the changes and pushed it, and now it randomly hangs on the CI server for clang 17, g++ and emsdk. They all work fine on my local machine. No idea either.